components/api/db/cellery_hub_db.bal

```diff
@@ -340,32 +389,95 @@ returns json | error {
     return check json.convert(imagesListResponse);
 }
 
-public function updateImage(string orgName, string imageName, string description, string userId) returns sql:UpdateResult | error? {
-    log:printInfo(io:sprintf("Performing update on image : %s in organization %s", imageName, orgName));
-    sql:UpdateResult res = check connection->update(UPDATE_IMAGE_QUERY, description, imageName, orgName, userId);
+public function updateImageDescriptionNSummary(string orgName, string imageName, string description, string summary, string userId) returns sql:UpdateResult | error? {
+    log:printInfo(io:sprintf("Updating description and summary of the image %s/%s", orgName, imageName));
+    sql:UpdateResult res = check connection->update(UPDATE_IMAGE_DESCRIPTION_N_SUMMARY_QUERY, description, summary, imageName, orgName, userId);
     return res;
 }
 
+public function updateImageKeywords(string orgName, string imageName, string[] keywords, string userId) returns error? {
+    log:printInfo(io:sprintf("Updating keywords of the image %s/%s", orgName, imageName));
+    string role = check getUserRole(userId, orgName);
+    if role == constants:ROLE_ADMIN || role == constants:ROLE_PUSH {
```

Comment:
This validation should not be done at this level. This should be done at the API level

Reply:
This validation is done in the above-mentioned query. Hence removing this validation.